moqt-core/src/modules/messages/control_messages/subscribe_ok.rs

```diff
@@ -47,40 +54,79 @@ impl MOQTPayload for SubscribeOk {
     where
         Self: Sized,
     {
-        let track_namespace_tuple_length = u8::try_from(read_variable_integer_from_buffer(buf)?)
-            .context("track namespace length")?;
-        let mut track_namespace_tuple: Vec<String> = Vec::new();
-        for _ in 0..track_namespace_tuple_length {
-            let track_namespace = String::from_utf8(read_variable_bytes_from_buffer(buf)?)
-                .context("track namespace")?;
-            track_namespace_tuple.push(track_namespace);
-        }
-        let track_name =
-            String::from_utf8(read_variable_bytes_from_buffer(buf)?).context("track name")?;
-        let track_id = read_variable_integer_from_buffer(buf).context("track id")?;
+        let subscribe_id = read_variable_integer_from_buffer(buf).context("subscribe_id")?;
         let expires = read_variable_integer_from_buffer(buf).context("expires")?;
+        let group_order_u8 = read_fixed_length_bytes_from_buffer(buf, 1)?[0];
+
+        // Values larger than 0x2 are a protocol error.
```

Comment:
Please use protocol violation instead of protocol error, since that's a term often used in drafts.